src/Sulu/Bundle/ContentBundle/Serializer/Subscriber/WorkflowStageSubscriber.php

```diff
@@ -0,0 +1,51 @@
+<?php
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\ContentBundle\Serializer\Subscriber;
+
+use JMS\Serializer\EventDispatcher\Events;
+use JMS\Serializer\EventDispatcher\EventSubscriberInterface;
+use JMS\Serializer\EventDispatcher\ObjectEvent;
+use Sulu\Component\Content\Document\Behavior\RedirectTypeBehavior;
+use Sulu\Component\Content\Document\Behavior\WorkflowStageBehavior;
+use Sulu\Component\Content\Document\WorkflowStage;
+
+/**
+ * Adds information about the workflow stage to the serialized version of the document.
+ */
+class WorkflowStageSubscriber implements EventSubscriberInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public static function getSubscribedEvents()
+    {
+        return [
+            [
+                'event' => Events::POST_SERIALIZE,
+                'format' => 'json',
+                'method' => 'onPostSerialize',
+            ],
+        ];
+    }
+
+    public function onPostSerialize(ObjectEvent $event)
+    {
+        /** @var RedirectTypeBehavior $document */
+        $document = $event->getObject();
+
+        if (!$document instanceof WorkflowStageBehavior) {
+            return;
+        }
+
+        $visitor = $event->getVisitor();
+        $visitor->addData('publishedState', $document->getWorkflowStage() === WorkflowStage::PUBLISHED);
```

Comment:
these properties are all legacy i think? shouldnt we mark this classes as deprecated?